crates/bevy_ecs/src/system/system_piping.rs

```diff
@@ -232,6 +233,54 @@ pub mod adapter {
         res.unwrap()
     }
 
+    /// System adapter that utilizes the info! macro to print system information.
+    /// 
+    /// # Examples
+    /// 
+    /// ```
+    /// 
+    /// ```
+    pub fn info<T: Display>(In(data): In<T>) {
+        tracing::info!("{}", data);
+    }
+
+    /// System adapter that utilizes the debug! macro to debug system piping.
+    ///
+    /// # Examples
+    /// 
+    /// ```
+    /// 
+    /// ```
+    pub fn dbg<T: Debug>(In(data): In<T>) {
+        tracing::debug!("{:?}", data);
+    }
+
+    /// System adapter that utilizes the warn! macro.
+    /// 
+    /// # Examples
+    /// 
+    /// ```
+    /// 
+    /// ```
+    pub fn warn<T: Debug>(In(res): In<Option<T>>) {
+        if let Some(warn) = res {
+            tracing::warn!("{:?}", warn);
+        }
+    }
+
+    /// System adapter that utilizes the error! macro which useful for fallible systems that should print the error message in the case of an error.
+    /// 
+    /// # Examples
+    /// 
+    /// ```
+    /// 
+    /// ```
+    pub fn error<T, E: Debug>(In(res): In<Result<T, E>>) {
+        if let Err(error) = res {
+            tracing::error!("{:?}", error);
+        }
+    }
```

Comment:
What if the user wants to pipe an `Option<T>` instead of `Result<T, E>`?

Reply:
Convert to a result inside your system IMO. Or you know, make your own adapter: they're pretty easy!

Reply:
Is it possible too make ``error(..)`` generic over ``Option`` and ``Error``/``anyhow::Error``? Perhaps by implementing a trait for all ``Option`` and ``Error`` types within this context? Or is that a bad idea?

Reply:
Probably a poor idea: it will make a simple method much harder to read.